### ISSUE TITLE
Example doesn't work with newest version: AngularFireUploadTask.downloadURL(); returns undefined

### ISSUE THREAD

**Issue by takahser** [angular/angularfire2#1645]
### Version info

**Angular: "^5.2.1"**
**Firebase: "^5.0.2"**
**AngularFire: "5.0.0-rc.7"**

**It used to work on earlier versions, e.g.:**
_"angularfire2": "5.0.0-rc.6",
"firebase": "^5.0.2",_

### How to reproduce these conditions
See the example [Monitoring upload percentage](https://github.com/angular/angularfire2/blob/74208a9b7eca1b7c64e0e24878aba95512f74447/docs/storage/storage.md#monitoring-upload-percentage)

```
@Component({
  selector: 'app-root',
  template: `
    <input type="file" (change)="uploadFile($event)" />
    <div>{{ uploadPercent | async }}</div>
    <a [href]="downloadURL | async">{{ downloadURL | async }}</a>
 `
})
export class AppComponent {
  uploadPercent: Observable<number>;
  downloadURL: Observable<string>;
  constructor(private storage: AngularFireStorage) {}
  uploadFile(event) {
    const file = event.target.files[0];
    const filePath = 'name-your-file-path-here';
    const task = this.storage.upload(filePath, file);
    
    // observe percentage changes
    this.uploadPercent = task.percentageChanges();
    // get notified when the download URL is available
    this.downloadURL = task.downloadURL();
  }
}
```

### Expected behavior
On the current version, when uploading a file using `AngularFireStorage.upload(filePath, file)` the downloadUrl of the file can be requested by subscribing to `AngularFireUploadTask.downloadURL()`:

`AngularFireUploadTask.downloadURL().subscribe(console.log) // should log download url`

### Actual behavior
Instead of returning the URL, the subscription returns `undefined`:

`AngularFireUploadTask.downloadURL().subscribe(console.log) // logs undefined`

**Comment by jamesdaniels** [comment]
Firebase 5.0 removed downloadUrl from upload task, so we did so too. I'll rework the doc.

**Comment by takahser** [comment]
@jamesdaniels thanks for the info. Did they replace it with something else or can't we use this functionality anymore?

**Comment by jamesdaniels** [comment]
You need to use downloadUrl on the ref.

**Comment by takahser** [comment]
@jamesdaniels thanks for your advice, this is actually what I tried however, in my case I'm uploading a file and I'd like to get the download url as soon as the file is uploaded. Even if I wait with requesting the file after it was uploaded (I tried wait for the `AngularFireStorage.upload` to be fulfilled), we have race conditions, since at this time the firebase storage service won't necessarily have processed the new image and therefore I get a 404. The only way I was able to make it work in a "reliable" way was by using `setTimeout` which is obviously very ugly... Is there a better way to do this?
